src/generate/handlebarsAmfHelpers.ts

```diff
@@ -0,0 +1,215 @@
+ * Get data type of a parameter
+ *
+ * @param param - instance of model.domain.Parameter
+ * @returns data type if defined in the parameter otherwise returns a default type
+ */
+export const getParameterDataType = (param: model.domain.Parameter): string => {
+  if (param != null && param.schema != null) {
+    return getDataType(param.schema);
+  }
+  return DEFAULT_DATA_TYPE;
+};
+
+/**
+ * Get type of the request body
+ *
+ * @param request - AMF model of tge request
+ * @returns Type of the request body
+ */
+export const getRequestPayloadType = (
+  request: model.domain.Request
+): string => {
+  if (
+    request != null &&
+    request.payloads != null &&
+    request.payloads.length > 0
+  ) {
+    const payloadSchema: model.domain.Shape = request.payloads[0].schema;
+    if (payloadSchema instanceof model.domain.ArrayShape) {
+      return ARRAY_DATA_TYPE.concat("<")
+        .concat(getPayloadType(payloadSchema.items))
+        .concat(">");
+    }
+    return getPayloadType(payloadSchema);
+  }
+  return OBJECT_DATA_TYPE;
+};
+
+/**
+ * Gets all properties of the DTO
+ *
+ * @param dtoTypeModel - AMF model of the dto
+ * @returns Array of properties in the dto that are not regular expressions
+ */
+export const getProperties = (
+  dtoTypeModel: model.domain.NodeShape | undefined | null
+): model.domain.PropertyShape[] => {
+  return getFilteredProperties(dtoTypeModel, propertyName => {
+    return !/^([/^]).*.$/.test(propertyName);
+  });
+};
+
+/**
+ * Check if the property is defined as required.
+ * Required properties have minimum count of at least 1
+ * We ignore required additional properties because of the
+ * different semantics used in rendering those properties
+ *
+ * @param property -
+ * @returns true if the property is required
+ */
+export const isRequiredProperty = (
+  property: model.domain.PropertyShape
+): boolean => {
+  return property != null && property.minCount.value() > 0;
+};
+
+/**
+ * Check if the property is optional.
+ * Optional properties have minimum count of 0
+ * We ignore optional additional properties which also have minimum count of 0,
+ * because of the different semantics used in rendering those properties.
+ *
+ * @param property -
+ * @returns true if the property is optional
+ */
+export const isOptionalProperty = (
+  property: model.domain.PropertyShape
+): boolean => {
+  return property != null && property.minCount.value() == 0;
+};
+
+/**
+ * Returns whether additional properties are allowed for a given RAML type.
+ *
+ * @param ramlTypeDefinition - Any RAML type definition
+ * @returns true if additional properties are allowed, false otherwise
+ */
+export const isAdditionalPropertiesAllowed = (
+  ramlTypeDefinition: model.domain.NodeShape
+): boolean => {
+  return (
+    ramlTypeDefinition !== undefined &&
+    ramlTypeDefinition.closed !== undefined &&
+    ramlTypeDefinition.closed.value !== undefined &&
+    !ramlTypeDefinition.closed.value()
+  );
+};
+
+const helpers = {
```

Comment:
I think the cleaner way to do this is just move the handlebars config into its own file. Then you can simply register everything in this file to it (and so can other people). I also prefer having a top-level package export of 'HandlebarsWithAmfHelpers' instead of it being nested inside the helpers export.

Reply:
Also, this appears to just be a list of keys with no values which means that nothing gets registered below :/

Reply:
Will move the config to a separate file.

ES6 allows just using values if they have the same names as the keys - https://alligator.io/js/object-property-shorthand-es6/

Reply:
You are right, I always forget that one. But my point is that you don't need the config at all.